### PR TITLE
Move test package logic to separate class, and add validation

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review.

### CODE REVIEW

**Comment by jonmv** [comment]
PTAL :)

**Review by freva — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 1]
```diff
@@ -907,33 +894,20 @@ private boolean useTesterCertificate(RunId id) {
...
+                                                  controller.controllerConfig().steprunner().testerapp(),
+                                                  spec,
+                                                  useTesterCertificate ? controller.clock().instant() : null,
+                                                  timeouts.testerCertificate());
+
+        if (useTesterCertificate) controller.jobController().storeTesterCertificate(id, testPackage.certificate());
+        return testPackage.asApplicationPackage();
     }
 
     private void appendAndStoreCertificate(ZipBuilder zipBuilder, RunId id) {
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java:913]: Thanks.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/notification/Notification.java [hunk 2]
```diff
@@ -66,10 +66,13 @@ public enum Level {
     }
 
     public enum Type {
-        /** Related to contents of application package, e.g. usage of deprecated features/syntax */
+        /** Related to contents of application package, e.g., usage of deprecated features/syntax */
         applicationPackage,
 
-        /** Related to deployment of application, e.g. system test failure, node allocation failure, internal errors, etc. */
+        /** Related to contents of application test package, e.g., mismatch between deployment spec and provided tests */
+        testPackage,
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/notification/Notification.java:73]: Yes, I assumed that.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/TestPackage.java [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+                         ", so it should also include " + (suites.contains(staging) ? "staging setup" : "staging tests"));
+        else if (expectedSuites.contains(staging) && ! suites.contains(staging))
+            problems.add("test package has no staging setup and tests, but <staging /> is declared in deployment.xml");
+
+        if (suites.contains(production) != expectedSuites.contains(production))
+            problems.add("test package has " + (suites.contains(production) ? "" : "no ") + "production tests, " +
+                         "but " + (suites.contains(production) ? "no " : "") + "production tests are declared in deployment.xml");
+
+        if ( ! problems.isEmpty())
+            problems.add("see https://docs.vespa.ai/en/testing.html for details on how to write system tests for Vespa");
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/application/pkg/TestPackage.java:194]: Sure.

**Pull request merged**